internal/fileutils/fileutils.go

```diff
@@ -572,6 +578,17 @@ func CopyAndRenameFiles(fromPath, toPath string) error {
 	return nil
 }
 
+func copyPath(info os.FileInfo, fromPath, toPath string) error {
+	if info.IsDir() {
+		err := MkdirUnlessExists(toPath)
+		if err != nil {
+			return errs.Wrap(err, "MkdirUnlessExists %s failed", toPath)
+		}
+		return CopyFiles(fromPath, toPath)
+	}
+	return CopyFile(fromPath, toPath)
+}
```

Comment:
This looks dangerous. The `CopyAndRenameFiles` already enumerates all the files it needs to copy, and you're calling `CopyFiles` here, essentially turning all subdirs into a `CopyFiles` instruction, but while also still processing them through `CopyAndRenameFiles`.

What are you trying to solve here?

Reply:
When I started to use `CopyAndRenameFiles` in the update generator, this change was necessary as the function currently does not handle directories in the `fromPath`. I can revert these changes and file a follow-up story as I'm no longer using `CopyAndRenameFiles` in the update generator. It is only being used in the installer and specifically for copying the contents of the `bin` directory.

Reply:
Yes, please revert and file a story. I'm not fully grokking what's happening here, it's looking more complicated than it probably needs to be. Either way not point holding up this PR over if you don't need it.

Reply:
Story filed here: https://www.pivotaltracker.com/story/show/178026455